src/test/lib/optimizer/strategy/data_induced_predicate_rule_test.cpp

```diff
@@ -0,0 +1,128 @@
+#include "lib/optimizer/strategy/strategy_base_test.hpp"
+
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "optimizer/strategy/data_induced_predicate_rule.hpp"
+
+namespace hyrise {
+
+using namespace expression_functional;  // NOLINT(build/namespaces)
+
+class DataInducedPredicateRuleTest : public StrategyBaseTest {
```

Comment:
note here: I would like to see more test cases where nothing is rewritten, e.g., due to the join mode. Furthermore, your code can rewrite both sides of a join, which is also an interesting test case.